index.js

```diff
@@ -86,20 +107,28 @@ class CMSClient {
         this.url = this.url.concat(`&$select=${this.fetchOptions.columns}`);
       }
     }
+
+    if (this.fetchOptions.order !== '') {
+      if (this.url[this.url.length - 1] === '?') {
+        this.url = this.url.concat(`$order=${this.fetchOptions.order}`);
+      } else {
+        this.url = this.url.concat(`&$order=${this.fetchOptions.order}`);
+      }
+    }
   }
 
   async _fetchResource() {
     try {
       const resourceData = await fetch(this.url);
       const headers = await resourceData.headers;
 
-      this.fetched.fields = JSON.parse(headers.get('x-soda2-fields').split(','));
+      this.fetched.fields = JSON.parse(
+        headers.get('x-soda2-fields').split(','),
+      );
       this.isOutdated = headers.get('x-soda2-data-out-of-date');
       this.lastModified = headers.get('Last-Modified');
 
-      if (this.fetchOptions.includeMetadata) {
-        await this._fetchResourceMetadata();
-      }
+      await this._fetchResourceMetadata();
```

Comment:
I think we should fetch the metadata regardless. We delete the metadata only if not wanted. I agree regarding the logical flow. Would it be cleaner to get rid of every other conditional and just have the following on line 131?

```js
 await this._fetchResourceMetadata();
 if (!this.fetchOptions.includeMetadata) delete this.fetched.metadata;
```
Passes all tests

Reply:
Gotcha! Remind me the reasoning for fetching regardless? Otherwise, I think that flow works perfectly.

Reply:
The resource API does not provide any information other than the raw data. Even for the most basic info such as name and dataset year we have to fetch the metadata. I was hoping to use the metadata to abstract the two endpoints and produce a clean output. However, I see how it might be confusing specially if `includeMetadata` is passed as false.